CHANGELOG.md

```diff
@@ -9,6 +9,11 @@
 * Allow null arguments on query. ([#872](https://github.com/realm/realm-dart/pull/872)). Fixes [#871](https://github.com/realm/realm-dart/issues/871)
 * Previously removeAt did not truncate length. ([#884](https://github.com/realm/realm-dart/pull/884)). Fixes [#883](https://github.com/realm/realm-dart/issues/883)
 * List.length= now throws, if you try to increase length, ([#894](https://github.com/realm/realm-dart/pull/894)).
+* Queries on results didn't filter the existing results. Ex:
+  ```dart
+  expect(realm.query<Person>('FALSEPREDICATE').query('TRUEPREDICATE'), isEmpty); // would fail if any Persons exists
+  ```
+  Fixes issue [#908](https://github.com/realm/realm-dart/issues/908)
```

Comment:
I think we are diverging from our changelog format. Why do we list the issue here? Previously we just list the PR with the change. In that PR there should be a link to the issue.

Reply:
I believe in all other repos we're listing the issue rather than the PR. There are two main reasons:
1. Users who file bug reports would like to see which release their bug was fixed in.
2. A single issue may be implemented over multiple PRs.

I don't know if that's as relevant today since Github now detects the release that includes a fix, but that was the reason before.

Reply:
I previously found this documented in:

According to https://wiki.corp.mongodb.com/display/REALM/Release+Notes+and+changelogs:
> Fixed
> Describe how the bug can be observed or provoked by the end-user and the consequences. Do not describe any implementation details. You can do that in the internal section or keep that in the issue/PR you reference.
> 
> If possible with reasonable effort (git blame) mention which version the bug was introduced in. This helps the end-user understand if they are at all affected by this bug.
> 
> General format:
> 
> * <when doing this, this could happen>. <Thanks to @external-contributor!> ([#_issue_](_url_), since v?)

So according to that we should link to the issue (not the PR) and describe how to hit it according to the above.

Personally I don't mind if we link it to the PR or the issue, I just aligned with what @nirinchev  suggested on my list related pr #906. 

Wrt. to the small sample that was added:
```dart
expect(realm.query<Person>('FALSEPREDICATE').query('TRUEPREDICATE'), isEmpty); // would fail if any Persons exists
```
I personally find it a bit overkill, as the user can just follow the link to the issue and get the same information, but I agree it has value that the users can evaluate if the fixed bug may/or may not impact them without traversing the link.

Reply:
Ok lets list the issue. Then I have a formatting suggestion. Will try to do that in code

Reply:
```suggestion
  * Queries on results didn't filter the existing results. ([#908](https://github.com/realm/realm-dart/issues/908))
  ```dart
  expect(realm.query<Person>('FALSEPREDICATE').query('TRUEPREDICATE'), isEmpty); // would fail if any Persons exists
  ```
```